flow/record/base.py

```diff
@@ -83,27 +108,28 @@ def _unpack(__cls, {args}):
 """
 
 
-class Peekable:
+class Peekable(BinaryIO):
```

Comment:
I wonder if [io.BufferedReader](https://docs.python.org/3/library/io.html#io.BufferedReader) could work instead of this custom class? Also, you'd probably want to subclass from `io.RawIOBase` instead of `BinaryIO`, which is a type hint.